src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs

```diff
@@ -213,15 +214,23 @@ internal IDependencyInjector DependencyInjector
             string operationName,
             Inputs inputs,
             IUserContext userContext,
+            ComplexityConfiguration complexityConfiguration,
             bool enableValidation = true,
             bool enableProfiling = false,
             IEnumerable<IValidationRule> rules = null,
             CancellationToken cancellationToken = default(CancellationToken))
+            
         {
             if (!enableValidation)
             {
                 rules = new[] { new NoopValidationRule() };
             }
+
+            complexityConfiguration = (query.ToLower().Contains("introspectionquery") &&
```

Comment:
Complexity validation can now simply be bypassed by a query like this:

```
query introspectionquery__schema {
	someQuery {
    someField
  }
}
```

Since the complexity configuration needs to be passed for each `Execute` (each query), I would leave that decision/detection up to the external calling code and remove this condition here.

In short, if the calling code wants to bypass the introspection query, then they are responsible for detecting introspection and not call `WithComplexityConfiguration` in that particular case.